### PR TITLE
feat(docs): add overview for integrating Netlify CSS with site generators

### PR DESCRIPTION
**Pull request by bytrangle**
**Summary**
This pull request contains a new Markdown file called Overview that is meant to be added to the Platform Guides in the docs. It is a high-level view of what it means to integrate Netlify CMS with common static site generators without being biased to any framework.

## Motivation
Integrating Netlify CMS into a website built with SSG contains many moving pieces. The process also requires a bit of a mental leap, especially for someone from monolithic CMS. There are already plenty of detailed guides for all the popular SSGs, but none really explains the process from a high level or the rationale behind each step.

With this article, I hope that readers will have a better grasp on the benefits of integrating Netlify CMS with their static sites, why they should do what they are instructed to do before diving head first into a specific SSG.

**Test plan**

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @bytrangle! I appreciate this effort in making it easier to integrate the CMS.

I've made some changes to the content, mostly to align wording and make it consistent.

Since we already have an [overview](https://www.netlifycms.org/docs/intro/) section that describes the CMS how about making the static site generator overview more specific to static site generators?

We would need to identify what information this overview needs to have, that is not included in other parts of the docs. Can you please suggest that?

Also if this PR contains information that is relevant for the general overview (such as backend authentication) maybe we should link https://www.netlifycms.org/docs/backends-overview/ from the general overview?

**To summarize** - the most important thing is to understand what makes this overview specific to static site generators, keep only that information in the overview and possibly add the rest to the general overview.

Please let me know if that makes sense

File: website/content/docs/site-generator-overview.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Netlify CMS doesn't store your site content. Think of it as a middleman that allows your CMS users to make changes to your Git repository without having an account with GitHub, GitLab etc.
+
+To achieve that, Netlify CMS needs to make sure that users have the write access to an online Git repository. This means the backend of the app is just JavaScript code that authenticates users with the Git provider which stores the site project.
+
+The authentication process requires a server and setting up a backend. You can write your own backend, or deploy your site to Netlify and get two authentication micro-services that simplify this process: Netlify Identity and Git Gateway.
+
+It is also helpful to remember that Netlify CMS works in production. Every time the repository changes, whether in the source codes or raw content, the website is rebuilt and redeployed automatically.
+
+For experimentation, you can run Netlify CMS locally, albeit with a catch. Each content edit through Netlify CMS results in a new commit to your remote repository. This means that you will need to pull changes from your remote each time to see them in the locally served site.
```
[the remark this answers is not in the document]
Comment (bytrangle) [reply website/content/docs/site-generator-overview.md:42]: Oh, it's a new feature? I remember doing several mental gymnastics when trying to make Netlify CMS app work with a local repo a few months ago. Should I delete both paragraphs and link to that beta feature?

[the remark this answers is not in the document]
Comment (erezrokah) [reply website/content/docs/site-generator-overview.md:42]: > Oh, it's a new feature?

A few months old I believe. Probably less visible since it's a beta feature.

> Should I delete both paragraphs and link to that beta feature?

Yes, that would be great!

File: website/content/docs/site-generator-overview.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The process for adding Netlify CMS to your site can vary between different generators. As a general approach, it has three main steps:
+
+1. Adding Netlify CMS to your site under the `/admin` route of the published site
+2. Setting up a backend to authenticate CMS users with the Git provider that hosts your source code
+3. Rendering the Netlify CMS generated content as web pages
+
+## Adding Netlify CMS to your site
+
+The frontend of the app contains the following major components:
```
[the remark this answers is not in the document]
Comment (bytrangle) [reply website/content/docs/site-generator-overview.md:24]: Yes, I'll change it to `application`.

File: website/content/docs/site-generator-overview.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Netlify CMS fills this gap. You can integrate it into any site built with a common static site generator, including but not limited to Gatsby, Hugo, Jekyll, Next.js, Nuxt.js, Middleman and Gridsome. This allows non-technical editors to create and edit content through a solid user interface, abstracting any Git related operations.
+
+The process for adding Netlify CMS to your site can vary between different generators. As a general approach, it has three main steps:
+
+1. Adding Netlify CMS to your site under the `/admin` route of the published site
+2. Setting up a backend to authenticate CMS users with the Git provider that hosts your source code
+3. Rendering the Netlify CMS generated content as web pages
+
+## Adding Netlify CMS to your site
```
[the remark this answers is not in the document]
Comment (bytrangle) [reply website/content/docs/site-generator-overview.md:22]: I meant to say that the very act of **adding Netlify CMS to a static site** is simply installing Netlify CMS as a React app at a route of a deployed site. *Installing* sounds more concrete than *adding* or *integrating*.

Rereading the section now, I realized that it didn't convey what I wanted to say. So I'm thinking of adding the following:
- Please refer to the dedicated instruction for your static site generator to see how to install Netlify CMS.
- An CMS app needs configuration. The configuration file for Netlify CMS is named `config.yml`. Then link to the dedication section for configuration file.

How does that sound?

File: website/content/docs/site-generator-overview.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Some site generators (e.g. Gatsby) come with plugins that simplify the installation of the Netlify CMS app.
+
+## Authenticate CMS users with a backend
+
+Netlify CMS doesn't store your site content. Think of it as a middleman that allows your CMS users to make changes to your Git repository without having an account with GitHub, GitLab etc.
+
+To achieve that, Netlify CMS needs to make sure that users have the write access to an online Git repository. This means the backend of the app is just JavaScript code that authenticates users with the Git provider which stores the site project.
+
+The authentication process requires a server and setting up a backend. You can write your own backend, or deploy your site to Netlify and get two authentication micro-services that simplify this process: Netlify Identity and Git Gateway.
```
[the remark this answers is not in the document]
Comment (bytrangle) [reply website/content/docs/site-generator-overview.md:38]: Should I rewrite it like below: 

“...deploy your site to Netlify and utilize two built-in tools that simplify this process:
- Netlify Identity, an authentication service
- Git Gateway, a server that lets you manage role-based access to your Git provider's API.
”

**Comment by bytrangle** [comment]
@erezrokah Currently, the SSG overview I wrote contains three sections:
- Installing Netlify CMS app.
- Authenticate users with  a backend
- Render Netlify CMS output on the frontend.

All of the three concepts above are not explored in detail in the General Overview.

I'm trying to think that if someone wants to add Netlify CMS to her deployed static site, what does the most streamlined process would look like? If she hasn't even finished putting together a static site, then she'd better off checking the documentation for a SSG, or  the SSG-specific section on Netlify CMS docs.

I assume that readers of this section already has static site, either locally or in production, because adding a CMS always comes after setting up a SSG. This also means that they already have a repo and familiar with pushing to a remote.

One thing I may add to the SSG overview is *Push to Git provider*, after *Installing Netlify CMS*. What happens after installing Netlify CMS? Push the changes, then redeploy the site. Now, users can check out the Netlify CMS app at `/admin`, although they won't be able to do anything else at this point. Then comes the *Authenticate CMS users*.

Do you have any suggestion?

File: website/content/docs/site-generator-overview.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply website/content/docs/site-generator-overview.md:44]: Yes, this looks great 🚀

File: website/content/docs/site-generator-overview.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply website/content/docs/site-generator-overview.md:22]: Thanks @bytrangle for clarifying. You suggestion sounds good

**Comment by erezrokah** [comment]
Hi @bytrangle, I'll try to explain myself better.

> * Installing Netlify CMS app.
> * Authenticate users with  a backend

These two topics are not related to static site generators, so I would expect them to be under the `Intro to Netlify CMS` section (the authentication part can also be under `Account Settings`)

The only core concept that I think is related to static site generators is

> * Render Netlify CMS output on the frontend.

as each SSG has a different way to generate HTML files from content files.

I understand now that your purpose is not to create a step by step guide, but a general section that explains core concepts. Let me know if I'm mistaken.
I think part of the reason for that is that our overview doesn't cover these core concepts.
I would like to see us improve the current overview to make that more clear.

For example, we could link to the installation guide and authentication section from here:
image

I hope that explains things better

File: website/content/docs/site-generator-overview.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+---
+title: Overview
+group: Guides
+weight: 1
+---
+
+A common pain point with static site generators is that they are hard to use for non-technical editors. Content and source code are stored together in the same Git repository. Making changes to the site content requires editors to go through a sequence of steps:
```
Comment (erezrokah) [inline website/content/docs/site-generator-overview.md:7]: I believe this part until `Adding Netlify CMS to your site` has a very similar intention to the [general overview](https://www.netlifycms.org/docs/intro/), so we might be able to combine the two

Comment (bytrangle) [reply website/content/docs/site-generator-overview.md:7]: What do you suggest? Move this part to the [general overview](https://www.netlifycms.org/docs/intro/), or delete it?

**Comment by erezrokah** [comment]
@bytrangle this looks great 👍  I made some minor changes in https://github.com/netlify/netlify-cms/pull/5342/commits/55211c41e5c3390e35080c0e9087669f07713ac1 (mostly to use relative links so they work in deploy previews and local development).

Please let me know if those changes look good and we can merge this PR.

**Review by erezrokah — APPROVED** [review]

**Comment by erezrokah** [comment]
Going to merge this. @bytrangle, if you have any comments we can follow up in another PR.
